File: lib/armstrong_numbers.js
/* An Armstrong number is an n-digit number that is equal to the sum of the nth
powers of its digits. */

var armstrongNumber = function(number){
  var stringOfNumber = number.toString();
  var exponent = stringOfNumber.length;
  var sumOfDigits = 0;
  for(var i = 0; i < exponent; ++i){
    sumOfDigits += Math.pow(parseInt(stringOfNumber[i]), exponent);   
  }
  if(sumOfDigits === number){
    console.log('true');
    return true;
  }
  else {
    console.log('false');
    return false;
  }
}

armstrongNumber(5);
armstrongNumber(153);
armstrongNumber(370);

armstrongNumber(25);
armstrongNumber(302);

File: lib/armstrong_numbers.rb
def armstrong_number?(num)
  string = num.to_s 
  n = string.length 
  digits = string.split(//) 
  sum = digits.inject(0) {|total, each| total + each.to_i ** n} 
  sum == num ? true : false 
end

armstrong_number?(153)
